mvpa2/clfs/gnb.py

```diff
@@ -332,16 +331,26 @@ def _call(self, dataset):
         pairs = list(itertools.combinations(range(len(clf.ulabels)), 2))
 
         weights = np.zeros([len(pairs), nfeat])
-        # do not compute sensitivity for features with variance 0 as this would implicate
-        # a division by zero
-        nonzero_vars = clf.variances!=0
+        # do not compute sensitivity for features with variance 0 as this would
+        # implicate a division by zero
+        nonzero_vars = clf.variances != 0
+        assert clf.params.common_variance
+        nonzero_vars0 = nonzero_vars[0, :]
         for idx, pair in enumerate(pairs):
-            weights[idx, nonzero_vars[0, :]] = (means[pair[0], nonzero_vars[0, :]] -
-                                                  means[pair[1], nonzero_vars[0, :]]) / \
-                                                 clf.variances[pair[0], nonzero_vars[0, :]]
+            # two-class sensitivity for (L0, L1) assumes that L1 is the
+            # "positive one"
+            weights[idx, nonzero_vars0] = (means[pair[1], nonzero_vars0] -
+                                           means[pair[0], nonzero_vars0]) / \
+                                          clf.variances[pair[0], nonzero_vars0]
 
```

Comment:
@yarikoptic I'm double checking sign of resulting beta estimates. Could you check whether I think correct here, just so that I'm not missing something obvious?

Situation: GLM on sensitivities (FFA-PPA) yields negative values for faces (as in plot seen before).

``pairs = list(itertools.combinations(range(len(clf.ulabels)), 2))`` will yield tuples [emitted in lexicographic sort order](https://docs.python.org/2/library/itertools.html), i.e. will have [('FFA', 'PPA'), ('EBA', 'FFA'), ...] instead of [('PPA', 'FFA'), ('FFA', 'EBA'), ...] and so forth.

```
weights[idx, nonzero_vars0] = (means[pair[1], nonzero_vars0] -
                               means[pair[0], nonzero_vars0]) / \
                               clf.variances[pair[0], nonzero_vars0]
```
will in the case of the ('FFA', 'PPA') pairs hence subtract FFA (index 0) estimates from PPA (index 1) estimates. If the FFA has a higher activation (mean) in a feature in question (e.g. faces) than the PPA, this would result in a negative feature weight for faces in the FFA-PPA decision boundary. 
Is that correct or am I missing something obvious?

Reply:
> pairs = list(itertools.combinations(range(len(clf.ulabels)), 2)) will yield tuples emitted in lexicographic sort order, i.e. will have [('FFA', 'PPA'), ('EBA', 'FFA'), ...] instead of [('PPA', 'FFA'), ('FFA', 'EBA'), ...] and so forth.

note the `range` in original invocation.  So here is no labels, just indexes to the labels.  `.ulabels` though should indeed be sorted but I would not assume it.  So if you have two labels

```
In [6]: list(itertools.combinations(range(2), 2))
Out[6]: [(0, 1)]
```
and if you have 3:
``
In [5]: list(itertools.combinations(range(3), 2))
Out[5]: [(0, 1), (0, 2), (1, 2)]
```
so you can see that the second index growing 'faster' so I think that neither of the two of your examples are correct ;) "lexicographic sort order" means here that those pairs are ordered in the order of original position(s) not that they are sorted alphabetically within entries:
```
In [7]: list(itertools.combinations('ABC', 2))
Out[7]: [('A', 'B'), ('A', 'C'), ('B', 'C')]

In [8]: list(itertools.combinations('CBA', 2))
Out[8]: [('C', 'B'), ('C', 'A'), ('B', 'A')]
```

Reply:
ahh crap, thanks for helping :)